cmd/kubeadm/app/cmd/join.go

```diff
@@ -71,18 +71,35 @@ func NewCmdJoin(out io.Writer) *cobra.Command {
 		},
 	}
 
-	cmd.PersistentFlags().StringVar(&cfgPath, "config", cfgPath, "Path to kubeadm config file")
+	cmd.PersistentFlags().StringVar(
+		&cfgPath, "config", cfgPath,
+		"Path to kubeadm config file")
+
+	cmd.PersistentFlags().StringVar(
+		&cfg.DiscoveryURL, "discovery-url", "",
+		"An HTTPS url from which to download cluster information")
+	cmd.PersistentFlags().StringVar(
+		&cfg.DiscoveryFile, "discovery-file", "",
+		"A file from which to load cluster information")
+	cmd.PersistentFlags().StringVar(
+		&cfg.DiscoveryToken, "discovery-token", "",
+		"A token used to validate cluster information fetched from the master")
+	cmd.PersistentFlags().StringVar(
+		&cfg.TLSBootstrapToken, "tls-bootstrap-token", "",
+		"A token used for TLS bootstrapping")
+	cmd.PersistentFlags().StringVar(
+		&cfg.Token, "token", "",
+		"Use this token for both discovery-token and tls-bootstrap-token")
+	cmd.PersistentFlags().StringSliceVar(
```

Comment:
Responded on main PR thread.  And the main documentation for the command currently states `join <master address>`.  We should update that with `join <args> [API server]` and the long description (there isn't one now) for the command should include "The API server on the command line must only be specified when using the token discovery scheme (set via --token or --discovery-token)."